### PR TITLE
fix: Do not throw NPE for UI.access called when session has expired

### LINKED ISSUE #12100: NullPointerException: Cannot invoke "com.vaadin.flow.server.VaadinSession.getErrorHandler()" logged when session is closed
### Description of the bug / feature
If your session times out, you may get the following exception:
```
com.vaadin.flow.component.UI             : Cannot invoke "com.vaadin.flow.server.VaadinSession.getErrorHandler()" because the return value of "com.vaadin.flow.component.UI.getSession()" is null

java.lang.NullPointerException: Cannot invoke "com.vaadin.flow.server.VaadinSession.getErrorHandler()" because the return value of "com.vaadin.flow.component.UI.getSession()" is null
	at com.vaadin.flow.component.UI$2.handleError(UI.java:531) ~[flow-server-8.0.2.jar:8.0.2]
	at com.vaadin.flow.server.FutureAccess.handleError(FutureAccess.java:76) ~[flow-server-8.0.2.jar:8.0.2]
	at com.vaadin.flow.server.VaadinService.runPendingAccessTasks(VaadinService.java:2045) ~[flow-server-8.0.2.jar:8.0.2]
	at com.vaadin.flow.server.VaadinSession.unlock(VaadinSession.java:709) ~[flow-server-8.0.2.jar:8.0.2]
	at com.vaadin.flow.server.VaadinService.ensureAccessQueuePurged(VaadinService.java:2005) ~[flow-server-8.0.2.jar:8.0.2]
	at com.vaadin.flow.server.VaadinService.accessSession(VaadinService.java:1972) ~[flow-server-8.0.2.jar:8.0.2]
	at com.vaadin.flow.server.VaadinSession.access(VaadinSession.java:1012) ~[flow-server-8.0.2.jar:8.0.2]
	at com.vaadin.flow.server.VaadinService.fireSessionDestroy(VaadinService.java:632) ~[flow-server-8.0.2.jar:8.0.2]
	at com.vaadin.flow.server.VaadinSession.valueUnbound(VaadinSession.java:204) ~[flow-server-8.0.2.jar:8.0.2]
	at org.apache.catalina.session.StandardSession.removeAttributeInternal(StandardSession.java:1860) ~[tomcat-embed-core-9.0.52.jar:9.0.52]
	at org.apache.catalina.session.StandardSession.expire(StandardSession.java:861) ~[tomcat-embed-core-9.0.52.jar:9.0.52]
	at org.apache.catalina.session.StandardSession.isValid(StandardSession.java:660) ~[tomcat-embed-core-9.0.52.jar:9.0.52]
	at org.apache.catalina.session.ManagerBase.processExpires(ManagerBase.java:608) ~[tomcat-embed-core-9.0.52.jar:9.0.52]
	at org.apache.catalina.session.ManagerBase.backgroundProcess(ManagerBase.java:591) ~[tomcat-embed-core-9.0.52.jar:9.0.52]
	at org.apache.catalina.core.StandardContext.backgroundProcess(StandardContext.java:5622) ~[tomcat-embed-core-9.0.52.jar:9.0.52]
	at org.apache.catalina.core.ContainerBase$ContainerBackgroundProcessor.processChildren(ContainerBase.java:1365) ~[tomcat-embed-core-9.0.52.jar:9.0.52]
	at org.apache.catalina.core.ContainerBase$ContainerBackgroundProcessor.processChildren(ContainerBase.java:1369) ~[tomcat-embed-core-9.0.52.jar:9.0.52]
	at org.apache.catalina.core.ContainerBase$ContainerBackgroundProcessor.processChildren(ContainerBase.java:1369) ~[tomcat-embed-core-9.0.52.jar:9.0.52]
	at org.apache.catalina.core.ContainerBase$ContainerBackgroundProcessor.run(ContainerBase.java:1347) ~[tomcat-embed-core-9.0.52.jar:9.0.52]
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539) ~[na:na]
	at java.base/java.util.concurrent.FutureTask.runAndReset(FutureTask.java:305) ~[na:na]
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:305) ~[na:na]
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) ~[na:na]
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) ~[na:na]
	at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61) ~[tomcat-embed-core-9.0.52.jar:9.0.52]
	at java.base/java.lang.Thread.run(Thread.java:833) ~[na:na]
```

This is due to `getSession()` returning `null`. 
### Minimal reproducible example
(fresh project from start.vaadin.com)
[status-dashboard.zip](https://github.com/vaadin/flow/files/7380170/status-dashboard.zip)
### Expected behavior
Log should not show a NullPointerException but either nothing or a sensible message stating that the session has expired or somethingsimilar. 
### Actual behavior
NPE is shown in the log.
### Versions:
    - Vaadin / Flow version: Vaadin 21.0.2
    - Java version: 17
    - OS version: Windows 10
    - Browser version (if applicable): n/a
    - Application Server (if applicable): Spring Boot embedded Tomcat
    - IDE (if applicable): n/a

### PR DESCRIPTION
**Pull request by Artur-**
Fixes #12100

### CODE REVIEW

File: flow-server/src/test/java/com/vaadin/flow/component/UITest.java [hunk 1]
```diff
@@ -536,6 +545,44 @@ public void unserSession_datachEventIsFiredForElements() {
...
     }
 
+    @Test
+    public void unsetSession_accessErrorHandlerStillWorks() throws IOException {
+        UI ui = createTestUI();
+
+        File output = File.createTempFile("logoutput", "uitest.log");
+        output.deleteOnExit();
+        System.setProperty("org.slf4j.simpleLogger.logFile",
+                output.getAbsolutePath());
```
Comment (caalador) [inline flow-server/src/test/java/com/vaadin/flow/component/UITest.java:555]: This doesn't sound too thread safe as some other test could add unexpected logs failing the test, so would it be better to use the `com.vaadin.flow.server.frontend.MockLogger` and make the `UI.getLogger` method package protected and overridable.

**Review by caalador — CHANGES_REQUESTED** [review]

File: flow-server/src/test/java/com/vaadin/flow/component/UITest.java (hunk 1 above)
Comment (Artur-) [reply flow-server/src/test/java/com/vaadin/flow/component/UITest.java]: Good point, makes the code better also

Comment (Artur-) [reply flow-server/src/test/java/com/vaadin/flow/component/UITest.java]: Done

**Review by caalador — APPROVED** [review]

**Comment by Artur-** [comment]
#12547

**Pull request merged**